src/ERC6551/ImpCookieJar6551.sol

```diff
@@ -29,6 +36,23 @@ contract ImpCookieJar6551 is CookieJarCore, Giver6551 {
         emit GiveCookie(cookieMonster, amount, CookieUtils.getCookieUid(POSTER_UID));
     }
 
+    function eatCookies(
+        uint256 amount,
+        address _cookieToken,
+        uint256 tokenId,
+        address nftContract) public {
```

Comment:
Does this means the caller now needs to find the NFT contract?

Reply:
yes, either that or the front end.  We can store the address in the contract on init if that makes things easier.  But i didn't want to mess with that.